streaming/base/storage/upload.py

```diff
@@ -616,6 +618,80 @@ def list_objects(self, prefix: Optional[str] = None) -> Optional[List[str]]:
         return []
 
 
+class HFUploader(CloudUploader):
+    """Upload file from local machine to a Huggingface Dataset.
+
+    Args:
+        out (str | Tuple[str, str]): Output dataset directory to save shard files.
+
+            1. If ``out`` is a local directory, shard files are saved locally.
+            2. If ``out`` is a remote directory then the shard files are uploaded to the
+               remote location.
+        keep_local (bool): If the dataset is uploaded, whether to keep the local dataset
+            shard file or remove it after uploading. Defaults to ``False``.
+        progress_bar (bool): Display TQDM progress bars for uploading output dataset files to
+            a remote location. Default to ``False``.
+        retry (int): Number of times to retry uploading a file. Defaults to ``2``.
+        exist_ok (bool): When exist_ok = False, raise error if the local part of ``out`` already
+            exists and has contents. Defaults to ``False``.
+    """
+
+    def __init__(self,
+                 out: str,
```

Comment:
Is it (str | Tuple[str, str])?

Reply:
No, I took the `Tuple` out. I can re-work this if we want it to be able to take a `Tuple`, but I couldn't figure out how the other classes were looping through them. 

HF does allow for downloading multiple files with a pattern (using `glob`) or downloading an entire folder, but otherwise we'd need to loop through each of the files in the Tuple to upload.

Reply:
@karan6181 see this comment (to consolidate on one thread)

Reply:
Yes, sgtm. 
please update the doc string with out (str): Output dataset directory to save shard files